src/pages/images/actions/forms/ImportImageFromTarballForm.tsx

```diff
@@ -0,0 +1,146 @@
+import { ChangeEvent, FC, useState } from "react";
+import { useEventQueue } from "context/eventQueue";
+import { useFormik } from "formik";
+import { useToastNotification } from "context/toastNotificationProvider";
+import {
+  changeImagePublicity,
+  createImageAlias,
+  createImageFromTarball,
+} from "api/images";
+import {
+  ActionButton,
+  Button,
+  Form,
+  Input,
+  Modal,
+} from "@canonical/react-components";
+import * as Yup from "yup";
+import { Link, useNavigate } from "react-router-dom";
+
+interface Props {
+  close: () => void;
+  project: string;
+}
+
+const ImportImageFromTarballForm: FC<Props> = ({ close, project }) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const navigate = useNavigate();
+  const [fileList, setFileList] = useState<FileList | null>(null);
+
+  const notifySuccess = () => {
+    const published = (
+      <Link to={`/ui/project/${project}/images`}>published</Link>
+    );
+    toastNotify.success(<>Image {published} from file.</>);
+  };
+
+  const changeFile = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files) {
+      const files = e.target.files;
+      setFileList(files);
+    }
+  };
+
+  const formik = useFormik<{
+    alias?: string;
+    isPublic: boolean;
+    fileList?: FileList;
+  }>({
+    initialValues: {
+      alias: "",
+      isPublic: false,
+    },
+    validationSchema: Yup.object().shape({
+      alias: Yup.string(),
+    }),
+    onSubmit: (values) => {
+      if (fileList) {
+        createImageFromTarball(fileList)
+          .then((operation) => {
+            toastNotify.info(<>Creation of image from file started.</>);
+            close();
```

Comment:
If we are tracking the file upload progress, then we should not close the modal here